sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/ExpressionEncoder.scala

```diff
@@ -67,6 +66,22 @@ object ExpressionEncoder {
       ClassTag[T](cls))
   }
 
+  // TODO: improve error message for java bean encoder.
+  def apply[T](beanClass: Class[T]): ExpressionEncoder[T] = {
```

Comment:
This is internal, but I probably wouldn't name this `apply` since we might want to support other types of classes with different rules in the future.